tutorials/iot/raspberrypi-wildlife-camera/raspberrypi-wildlife-camera.md

```diff
@@ -0,0 +1,591 @@
+
+Our motion detection is going to be based on _Motion_, a widely used open source project. In this step, we are going to pull motion from source, build it and add a custom configuration. We are not going to do it by hand, though. We are going to use snapcraft to package it as a snap, so our package will be modular, easy to tweak, test, install, share, and ultimately distribute.
+
+As an example, you will be able to tell people running almost any Linux distribution to run:
+
+``` bash
+snap install --channel=edge my-wildlife-camera-app
+```
+
+And they will get an automated update each time you change the source code.
+
+### Let's get started
+
+First, we need to initialize a new snapcraft project.
+
+Open a terminal and create a new directory called `wildlife-camera-tutorial` and enter it:
+
+``` bash
+mkdir wildlife-camera-tutorial
+cd wildlife-camera-tutorial
+```
+
+In this empty directory, run `snapcraft init` to initialize an new snapcraft project. This will create a `snap` directory containing a `snapcraft.yaml` file.
+
+### Adding general metadata
+
+This `snapcraft.yaml` file contains some boilerplate that we are going to change. Open it with your favourite text editor and at the top of the file, edit the existing values to match this:
+
+``` yaml
+name: wildlife-camera-tutorial
+version: 0.1
+summary: A wildlife motion-sensing camera app
+description: |
+ A wildlife camera app that will record small videos from a USB webcam
+ when motion is detected.
+grade: stable
+confinement: strict
+```
+
+You have just prepared the scene to create a snap. You have declared its:
+
+* `name`: a unique lowercase string that will identify your snap on the command-line and in the store
+* `version`: a human-readable version string so users know what they are running
+* `summary`: a short description appearing in store search results
+* `description`: a longer, multi-line description
+* `grade`: the state of the app. It can be `devel` (for software still in development) or `stable`, which will affect how it can be released to the store
+* `confinement`: our app is going to be strictly confined and won't be able to read or write anything on the host system outside of its confinement area
+
+### Adding a "motion" part
+
+Under the general metadata, we are going to add a "part". A declaration for a piece of code that will be included in our package.
+
+Add:
+
+``` yaml
+parts:
+  motion:
+    plugin: autotools
+    source: https://github.com/Motion-Project/motion.git
+    build-packages:
+      - libzip-dev
+      - libjpeg-dev
+      - libavformat-dev
+      - libavcodec-dev
+      - libavutil-dev
+      - libswscale-dev
+      - libavdevice-dev
+      - libwebp-dev
+      - pkg-config
+    stage-packages:
+      - gettext-base
+      - zlib1g
+      - libpulse0
+```
+
+This part we are calling "motion" declares how to build Motion from source. Without entering into too much details, here is what we have declared:
```

Comment:
If you're not familiar with snapcraft's use of *part* this could be a little confusing. eg. This defines a *part* we are calling motion, which declares...